src/logging.cpp

```diff
@@ -222,23 +226,43 @@ void BCLog::Logger::LogPrintStr(const std::string &str)
     if (m_print_to_file) {
         std::lock_guard<std::mutex> scoped_lock(m_file_mutex);
 
+        // While the system is initializing, save some reasonable
+        // number of log messages for re-logging when the log
+        // file rotates (gets reopened).
+        if (m_relog_enable && m_msgs_relog.size() < 1000) {
+            m_msgs_relog.push_back(str_prefixed);
+        }
+
         // buffer if we haven't opened the log yet
         if (m_fileout == nullptr) {
             m_msgs_before_open.push_back(str_prefixed);
         }
         else
         {
+            FileWriteStr(str_prefixed, m_fileout);
```

Comment:
This is only moved from below the `m_reopen_file` if statement below, see comment below.